server/src/server.rs

```diff
@@ -264,6 +264,19 @@ impl Server {
         Ok(())
     }
 
+    fn shutdown_threads(&mut self){
+        let shutdown_notification = Message::Notification(lsp_server::Notification{
+            method: Shutdown::METHOD.to_string(),
+            params: serde_json::Value::Null,
+        });
+        for _ in 0..self.senders_s_to_main.len() {
+            self.sender_s_to_main.send(shutdown_notification.clone()).unwrap(); //sent as notification as we already handled the request for the client
+        }
+        for _ in 0..self.senders_s_to_read.len() {
+            self.sender_s_to_read.send(shutdown_notification.clone()).unwrap(); //sent as notification as we already handled the request for the client
+        }
+        info!("Got shutdown request. Exiting.");
```

Comment:
as shutdown_threads is call from both client or server, Could reflect this information in the log?

Reply:
Updated!